include/CL/opencl.hpp

```diff
@@ -10253,7 +10370,118 @@ namespace compatibility {
     };
 } // namespace compatibility
 
+#ifdef cl_khr_semaphore
+class Semaphore : public detail::Wrapper<cl_semaphore_khr>
+{
+public:
+    Semaphore() : detail::Wrapper<cl_type>() {}
+    Semaphore(
+        const Context &context,
+        const vector<cl_semaphore_properties_khr>& sema_props,
+        cl_int *err = NULL) 
+    {
+        /* initialization of addresses to extension functions (it is done only once) */
+        std::call_once(ext_init_, initExtensions, context);
+
+        cl_int error = ext_init_error_;
+
+        if (error == CL_SUCCESS) {
+            object_ = pfn_clCreateSemaphoreWithPropertiesKHR(
+                context(), sema_props.data(), &error);
+        }        
+
+        detail::errHandler(error, __CREATE_SEMAPHORE_KHR_WITH_PROPERTIES_ERR);
+
+        if (err != NULL) {
+            *err = error;
+        }
+    }
+    Semaphore(
+        const vector<cl_semaphore_properties_khr>& sema_props,
+        cl_int* err = NULL):Semaphore(Context::getDefault(err), sema_props, err) {}
+    
+    explicit Semaphore(const cl_semaphore_khr& semaphore, bool retainObject = false) :
+        detail::Wrapper<cl_type>(semaphore, retainObject) {}
+    Semaphore& operator = (const cl_semaphore_khr& rhs) {
+        detail::Wrapper<cl_type>::operator=(rhs);
+        return *this;
+    }
+    template <typename T>
+    cl_int getInfo(cl_semaphore_info_khr name, T* param) const
+    {
+        typedef CL_API_ENTRY cl_int(CL_API_CALL * PFN_clGetSemaphoreInfoKHR)(
+            cl_semaphore_khr sema_object, cl_semaphore_info_khr param_name,
+            size_type param_value_size, void* param_value,
+            size_type* param_value_size_ret);
+
```

Comment:
I think this is leftover code and can be removed?

Reply:
Indeed, code removed. thank you!